topics/data-science/tutorials/bash-git/git_tutorial.md

```diff
@@ -0,0 +1,1928 @@
+---
+layout: tutorial hands-on
+title: "Version Control with Git"
+time_estimation: "65M"
+questions:
+- "What is version control and why should I use it?"
+- "How do I get set up to use Git?"
+- "Where does Git store information?"
+- "How do I record changes in Git?"
+- "How do I check the status of my version control repository?"
+- "How do I record notes about what changes I made and why?"
+- "How can I identify old versions of files?"
+- "How do I review my changes?"
+- "How can I recover old versions of files?"
+objectives:
+- "Understand the benefits of an automated version control system."
+- "Understand the basics of how automated version control systems work."
+- "Configure `git` the first time it is used on a computer."
+- "Understand the meaning of the `--global` configuration flag."
+- "Create a local Git repository."
+- "Describe the purpose of the `.git` directory."
+- "Go through the modify-add-commit cycle for one or more files."
+- "Explain where information is stored at each stage of that cycle."
+- "Distinguish between descriptive and non-descriptive commit messages."
+- "Explain what the HEAD of a repository is and how to use it."
+- "Identify and use Git commit numbers."
+- "Compare various versions of tracked files."
+- "Restore old versions of files."
+keypoints:
+- "Version control is like an unlimited 'undo'."
+- "Version control also allows many people to work in parallel."
+- "Use `git config` with the `--global` option to configure a user name, email address, editor, and other preferences once per machine."
+- "`git init` initializes a repository."
+- "Git stores all of its repository data in the `.git` directory."
+- "`git status` shows the status of a repository."
+- "Files can be stored in a project's working directory (which users see), the staging area (where the next commit is being built up) and the local repository (where commits are permanently recorded)."
+- "`git add` puts files in the staging area."
+- "`git commit` saves the staged content as a new commit in the local repository."
+- "Write a commit message that accurately describes your changes."
+- "`git diff` displays differences between commits."
+- "`git checkout` recovers old versions of files."
+contributors:
+  - Sofokli5
+  - fpsom
+notebook:
+  language: bash

```

Comment:
[*Comment, not something you need to change*]

Right now this won't generate a great notebook, because of the `code-in` blocks where students should write something. All of them would have to be converted to left aligned codeblocks, not in a blockquote.

But maybe that's a sign we need to improve the notebook engine to parse those, because they looks a lot nicer.

Reply:
We can try it out. Can you give a brief example of how code should be inserted, so that we can update as many as we can?

Reply:
just like https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#automatic-jupyter-notebooks--rmarkdown, i.e. without the `code-in`, but, I think this is a problem for me to solve. Currently the notebook parser doesn't recurse into blockquote blocks.